### DISCUSSION TITLE
Duckdb vs Redis

### DISCUSSION THREAD

**Discussion by saisaigraph** [duckdb/duckdb#5471] [General]
Could anyone pour some light when to use Redis vs Duckdb please.
Thanks
Sai

**Comment by Alex-Monahan** [comment]
Redis is key/value, whereas DuckDB is OLAP. That means DuckDB is better for aggregating or joining large volumes of data. Redis also runs as a separate process, which means many processes can access it, but DuckDB is faster for transferring data back and forth to an app. DuckDB does support indexing for fast individual lookups, but Redis is likely still much faster for that task. 

Does that help?

**Reply by saisaigraph** [reply]
Thank you Alex.

**Comment by saralanuj** [comment]
a) Sir, can DuckDB runs as a separate process (in memory) so that other processes can access it - just like Redis. Is that possible?
b) I would be getting streaming tick data for stock market (only 5 or 6 fields). I would be converting them to 1 min candle. Total max size will go from 0 to half a gb in 6 hours trading session. would that fit DuckDB usecase or Redis. If you can shade some light that will of immense help brother.

**Reply by Tishj** [reply]
You can build an application that allows other processes to communicate to the duckdb that runs on that process, I don't see why not, I think there was a third party that was writing a webserver duckdb extension that does something similar.

I can't answer b
